lsp/server/src/server.ts

```diff
@@ -34,92 +34,85 @@ const documents: TextDocuments<TextDocument> = new TextDocuments(TextDocument);
 let hasConfigurationCapability = false;
 let hasWorkspaceFolderCapability = false;
 let messDef:string;
-
+let allowConfigurationChanges = false;
 let startedSA4U_Z3 = false;
-connection.onInitialize((params: InitializeParams) => {
 
-	const capabilities = params.capabilities;
+const execAsync = promisify(exec);
 
-	// Does the client support the `workspace/configuration` request?
-	// If not, we fall back using global settings.
-	hasConfigurationCapability = !!(
-		capabilities.workspace && !!capabilities.workspace.configuration
-	);
-	hasWorkspaceFolderCapability = !!(
-		capabilities.workspace && !!capabilities.workspace.workspaceFolders
-	);
-
-	const result: InitializeResult = {
-		capabilities: {
-			codeActionProvider: true,
-			textDocumentSync: TextDocumentSyncKind.Incremental,
-			executeCommandProvider: {
-				commands: ['sa4u.fix']
-			}
-		}
-	};
-	if (hasWorkspaceFolderCapability) {
-		result.capabilities.workspace = {
-			workspaceFolders: {
-				supported: true
-			}
-		};
+class SA4UConfig {
+	compilationDir: string
+	constructor(compilationDir: string) {
+		this.compilationDir = '/src/' + compilationDir;
 	}
-	return result;
-});
+}
 
-connection.onInitialized(() => {
-	
-	if (hasConfigurationCapability) {
-		// Register for all configuration changes.
-		connection.client.register(DidChangeConfigurationNotification.type, undefined);
-		connection.workspace.getConfiguration('SA4U').then((value) => {
-			if (value.messageDefinition) {
-				messDef = value.messageDefinition;
-			}
-		});
-	}
-	if (hasWorkspaceFolderCapability) {
-		connection.workspace.onDidChangeWorkspaceFolders(_event => {
-			connection.console.log('Workspace folder change event received.');
-		});
-	}
-	connection.workspace.getConfiguration();
+async function checkForDockerRunningAndInstalled() {
```

Comment:
What type does this function return?

Reply:
Returns an error message to show to the user if a problem is found. Otherwise returns a resolved promise.